tools/github/write_current_source_version.py

```diff
@@ -68,10 +68,12 @@
     # Fetch the current version commit information from GitHub.
     commit_info_request = urllib.request.Request(
         "https://api.github.com/repos/envoyproxy/envoy/commits/v" + current_version)
-    github_token = os.environ.get(args.github_api_token_env_name)
-    if github_token != None:
+    if github_token := os.environ.get(args.github_api_token_env_name):
+        # Reference: https://github.com/octokit/auth-token.js/blob/902a172693d08de998250bf4d8acb1fdb22377a4/src/with-authorization-prefix.ts#L6-L12.
+        authorization_header_prefix = "bearer" if len(github_token.split(".")) is 3 else "token"
```

Comment:
```suggestion
        authorization_header_prefix = "bearer" if len(github_token.split(".")) == 3 else "token"
```

```console
F632 use ==/!= to compare constant literals (str, bytes, int, float, tuple)
```

i think our flake8 is a bit limited and needs to have more rules added